app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -156,17 +177,85 @@ public void updateIfNeeded() {
         Element options = (Element) mConfig.getDocumentElement()
                 .getElementsByTagName("options").item(0);
         changed = setConfigElement(options, "weakHashSelectionMethod", "never") || changed;
 
+        /* Dismiss "fsWatcherNotification" according to https://github.com/syncthing/syncthing-android/pull/1051 */
+        NodeList childNodes = options.getChildNodes();
+        for (int i = 0; i < childNodes.getLength(); i++) {
+            Node node = childNodes.item(i);
+            if (node.getNodeName().equals("unackedNotificationID")) {
+                if (node.equals("fsWatcherNotification")) {
+                    Log.i(TAG, "Remove found unackedNotificationID 'fsWatcherNotification'.");
+                    options.removeChild(node);
+                    changed = true;
+                    break;
+                }
+            }
+        }
+
+        // Save changes if we made any.
         if (changed) {
             saveChanges();
         }
     }
 
+    /**
+     * Updates syncthing options to a version specific target setting in the config file.
+     *
+     * Used for one-time config migration from a lower syncthing version to the current version.
+     * Enables filesystem watcher.
+     */
+    private boolean migrateSyncthingOptions () {
+        boolean changed = false;
+
+        /* Read existing config version */
+        int iConfigVersion = Integer.parseInt(mConfig.getDocumentElement().getAttribute("version"));
+        Log.i(TAG, "Found existing config version " + Integer.toString(iConfigVersion));
+
+        /* Check if we have to do manual migration from version X to Y */
+        if (iConfigVersion == 27) {
+            /* fsWatcher transition - https://github.com/syncthing/syncthing/issues/4882 */
+            Log.i(TAG, "Migrating config version " + Integer.toString(iConfigVersion) + " to 28 ...");
+
+            /* Enable fsWatcher for all folders */
+            NodeList folders = mConfig.getDocumentElement().getElementsByTagName("folder");
+            for (int i = 0; i < folders.getLength(); i++) {
+                Element r = (Element) folders.item(i);
+
+                // Enable "fsWatcherEnabled" attribute and set default delay.
+                Log.i(TAG, "Set 'fsWatcherEnabled', 'fsWatcherDelayS' on folder " + r.getAttribute("id"));
+                r.setAttribute("fsWatcherEnabled", Boolean.toString(true));
+                r.setAttribute("fsWatcherDelayS", Integer.toString(10));
+            }
+
+            /**
+            * Set config version to 28 after manual config migration
+            * This prevents "unackedNotificationID" getting populated
+            * with the fsWatcher GUI notification.
+            */
+            iConfigVersion = 28;
+            changed = true;
+        }
+
+        mConfig.getDocumentElement().setAttribute("version", Integer.toString(iConfigVersion));
+        Log.i(TAG, "New config version is " + Integer.toString(iConfigVersion));
```

Comment:
This could be under a `if (changed)` as otherwise it will always print new config version, which might not change in most cases.
The other (potentially better) option would be to have `iOldConfigVersion` and `iConfigVersion` and only update and print stuff if old does not match new.

Reply:
ok, I chose the "better" option.